internal/runtime/internal/importsource/import_git.go

```diff
@@ -194,11 +195,13 @@ func (im *ImportGit) Update(args component.Arguments) (err error) {
 	if im.repo == nil || !reflect.DeepEqual(repoOpts, im.repoOpts) {
 		r, err := vcs.NewGitRepo(context.Background(), repoPath, repoOpts)
 		if err != nil {
-			if errors.As(err, &vcs.UpdateFailedError{}) {
+			isUpdateFailedError := errors.As(err, &vcs.UpdateFailedError{})
+			if errors.Is(err, plumbing.ErrReferenceNotFound) || !isUpdateFailedError {
+				return err
+			}
+			if isUpdateFailedError {
```

Comment:
Oh, this "if" condition is to catch when `isUpdateFailedError` is true though :D

Reply:
When I commented the order was different but the idea was that if you check that isUpdateFailedError is false above and that you return, then in the code that follows isUpdateFailedError will be true

if A || !B {
 return
}
if B { <----- the condition is not needed because B will always be true since we return when B is false above
  foo()
}

This is not relevant anymore though